tests/TypeReconciliation/ReconcilerTest.php

```diff
@@ -177,4 +176,61 @@ public function providerTestTypeIsContainedBy(): array
             ],
         ];
     }
+
+    /**
+     * @dataProvider constantAssertions
+     */
+    public function testReconciliationOfClassConstantInAssertions(string $assertion, string $expected_type): void
+    {
+        $this->addFile(
+            'psalm-assert.php',
+            '
+            <?php
+            class Foo
```

Comment:
Can you add a test in a namespace please?

Reply:
What do you mean with "namespace"?
So rather than adding a file with `class Foo`, better add `namespace ReconciliationTest;` or sth. like this in there?

Reply:
what I meant is that your Foo class is in global namespace, when the class is in a namespace, some part of the code works differently, so yeah, please add a test with a `namespace test;` so that we can see everything works

Reply:
Okay so I understood you right 👍🏼